test/unit/html/HtmlConverterTest.php

```diff
@@ -9,103 +9,163 @@
-        $converter = new HtmlConverter($html, 'UTF-8', $store->settings['project_token'], $store, $headers);
-        list($translated_html, $marker) = $converter->insertSnippetAndHreflangTags(false);
+        $websites = array(
+            'real_html/stack_overflow',
+            'real_html/youtube',
+            'real_html/yelp',
+            'real_html/yahoo_jp'
+        );
 
-        $expected_html_text = file_get_contents('test/fixtures/real_html/yelp_expected.html');
-        $doc = new \DOMDocument("1.0", "ISO-8859-15");
-        $doc->loadHTML(mb_convert_encoding($expected_html_text, 'HTML-ENTITIES', "utf-8"));
-        $expected_html = $doc->saveHTML();
+        foreach ($websites as $website_name) {
+            // insert snippet and hreflang
+            $original_html = file_get_contents("test/fixtures/{$website_name}.html");
+            list($store, $headers) = StoreAndHeadersFactory::fromFixture('default');
+            $converter = new HtmlConverter('UTF-8', $store->settings['project_token'], $store, $headers);
+            $translated_html = $converter->insertSnippetAndHreflangTags($original_html, false);
 
-        $actual_html_text = $marker->revert($translated_html);
-        $doc = new \DOMDocument("1.0", "ISO-8859-15");
-        $doc->loadHTML(mb_convert_encoding($actual_html_text, 'HTML-ENTITIES', "utf-8"));
-        $actual_html = $doc->saveHTML();
+            $actual_html = $this->convertEncordingAndCorrectHtml($converter->revertMarkers($translated_html));
+            $expected_html = $this->convertEncordingAndCorrectHtml(file_get_contents("test/fixtures/{$website_name}_expected.html"));
 
-        $this->assertEquals($expected_html, $actual_html);
+            $this->assertEquals($expected_html, $actual_html);
+        }
     }
 
-    public function testConvertAndRevertAtYahooJp()
-    {
-        libxml_use_internal_errors(true);
-        $html = file_get_contents('test/fixtures/real_html/yahoo_jp.html');
-        list($store, $headers) = StoreAndHeadersFactory::fromFixture('default');
-        $converter = new HtmlConverter($html, 'UTF-8', $store->settings['project_token'], $store, $headers);
-        list($translated_html, $marker) = $converter->insertSnippetAndHreflangTags(false);
-
-        $expected_html_text = file_get_contents('test/fixtures/real_html/yahoo_jp_expected.html');
-        $doc = new \DOMDocument("1.0", "ISO-8859-15");
-        $doc->loadHTML(mb_convert_encoding($expected_html_text, 'HTML-ENTITIES', "utf-8"));
-        $expected_html = $doc->saveHTML();
-
-        $actual_html_text = $marker->revert($translated_html);
-        $doc = new \DOMDocument("1.0", "ISO-8859-15");
-        $doc->loadHTML(mb_convert_encoding($actual_html_text, 'HTML-ENTITIES', "utf-8"));
-        $actual_html = $doc->saveHTML();
-
-        $this->assertEquals($expected_html, $actual_html);
+    public function testInsertSnippetAndHreflangTags()
+    {
+        $html_cases = array(
```

Comment:
I think if you just want to re-use the code in the test, you can make a helper method for the common code. With this style of test you're basically making a "test framework inside of a test framework". It's already starting to get a bit complicated here so I would be careful.

Reply:
I want to do a parameterized test.
I also write as DRY.
When I modify code in WOVN.php, it is really hard.
Because there are many similar test cases, but sometimes there are no needed test cases.
I think a few function is easy to modify and compare test cases.